sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastNestedLoopJoin.scala

```diff
@@ -47,7 +47,7 @@ case class BroadcastNestedLoopJoin(
   override def outputsUnsafeRows: Boolean = left.outputsUnsafeRows || right.outputsUnsafeRows
   override def canProcessUnsafeRows: Boolean = true
 
-  @transient private[this] lazy val resultProjection: InternalRow => InternalRow = {
+  @transient private[this] def genResultProjection: InternalRow => InternalRow = {
```

Comment:
How does this work? Right now we generate the class on the driver and than ship the generated closure. Shouldn't we generate the class on the executors?